cmake/Conan.cmake

```diff
@@ -18,7 +18,7 @@ conan_cmake_run(
   ${CONAN_EXTRA_REQUIRES}
   catch2/2.11.0
```

Comment:
```suggestion
  catch2/2.12.1
```

Reply:
Thanks for the approval and suggestions, but for this PR I was really only interested in fixing the dependency warning. If the developers of spdlog think that their package requires a more up-to-date version of fmt, then there's probably a bug somewhere that's caused by pairing spdlog with the older version. That's the bug that this PR is meant to prevent.

You are welcome to make your own PR to update the other dependencies; that's probably a good idea. I would rather keep those changes separate from this PR though.